backend/infrahub/computed_attribute/models.py

```diff
@@ -69,15 +79,190 @@ class PythonTransformComputedAttribute:
+            trigger=event_trigger,
+            actions=[workflow],
+        )
+
+        return definition
+
+
+class ComputedAttrPythonTriggerDefinition(TriggerBranchDefinition):
+    type: TriggerType = TriggerType.COMPUTED_ATTR_PYTHON
+    computed_attribute: PythonTransformComputedAttribute
+
+    @classmethod
+    def from_object(
+        cls,
+        branch: str,
+        computed_attribute: PythonTransformComputedAttribute,
+        branches_out_of_scope: list[str] | None = None,
+    ) -> Self:
+        # scope = registry.default_branch
+
+        event_trigger = EventTrigger()
+        event_trigger.events.update({NodeCreatedEvent.event_name, NodeUpdatedEvent.event_name})
+        event_trigger.match = {
+            "infrahub.node.kind": [computed_attribute.computed_attribute.kind],
+        }
+
+        if branches_out_of_scope:
+            event_trigger.match["infrahub.branch.name"] = [f"!{branch}" for branch in branches_out_of_scope]
+        elif not branches_out_of_scope and branch != registry.default_branch:
+            event_trigger.match["infrahub.branch.name"] = branch
+
+        definition = cls(
+            name=computed_attribute.computed_attribute.key_name,
+            branch=branch,
+            computed_attribute=computed_attribute,
+            trigger=event_trigger,
+            actions=[
+                ExecuteWorkflow(
+                    workflow=COMPUTED_ATTRIBUTE_PROCESS_TRANSFORM,
+                    parameters={
+                        "branch_name": "{{ event.resource['infrahub.branch.name'] }}",
+                        "node_kind": "{{ event.resource['infrahub.node.kind'] }}",
+                        "object_id": "{{ event.resource['infrahub.node.id'] }}",
+                        "computed_attribute_name": computed_attribute.computed_attribute.attribute.name,
+                        "computed_attribute_kind": computed_attribute.computed_attribute.kind,
+                        "context": {
+                            "__prefect_kind": "json",
+                            "value": {
+                                "__prefect_kind": "jinja",
+                                "template": "{{ event.payload['context'] | tojson }}",
+                            },
+                        },
+                    },
+                ),
+            ],
+        )
+
+        return definition
+
+
+class ComputedAttrPythonQueryTriggerDefinition(TriggerBranchDefinition):
+    type: TriggerType = TriggerType.COMPUTED_ATTR_PYTHON_QUERY
+
+    @classmethod
+    def from_object(
+        cls,
+        branch: str,
+        computed_attribute: PythonTransformComputedAttribute,
+        branches_out_of_scope: list[str] | None = None,
+    ) -> Self:
+        event_trigger = EventTrigger()
+        event_trigger.events.update({NodeCreatedEvent.event_name, NodeUpdatedEvent.event_name})
+        event_trigger.match = {
+            "infrahub.node.kind": [computed_attribute.computed_attribute.kind],
```

Comment:
This part needs to be updated to include the impacted kinds for the query like this:

```diff
-            "infrahub.node.kind": [computed_attribute.computed_attribute.kind],
+            "infrahub.node.kind": computed_attribute.query_models,
```